### PR TITLE
a few minor cleanups

### PR DESCRIPTION
**Pull request by lzutao**
changelog: none

### CODE REVIEW

**Comment by flip1995** [comment]
r=me once travis passes

@bors delegate+

File: src/driver.rs [hunk 1]
```diff
@@ -385,23 +376,18 @@ pub fn main() {
...
+                args.extend(vec!["--cfg".into(), r#"feature="cargo-clippy""#.into()]);
                 if let Ok(extra_args) = env::var("CLIPPY_ARGS") {
-                    args.extend(extra_args.split("__CLIPPY_HACKERY__").filter_map(|s| {
-                        if s.is_empty() {
-                            None
-                        } else {
-                            Some(s.to_string())
-                        }
-                    }));
+                    match extra_args.strip_suffix("__CLIPPY_HACKERY__") {
```
Comment (flip1995) [inline src/driver.rs]: This isn't the same, is it? This assumes that `__CLIPPY_HACKERY__` is at the end of the string, while the prior version just assumes that it is anywhere in the string.

Comment (lzutao) [reply src/driver.rs]: removed.

**Pull request merged**